### PR TITLE
Add initial bindings to Core Animation.

### PR DESCRIPTION
**Pull request by pcwalton**
Bound types include `CALayer`, `CATransaction`, and `CATransform3D`.

r? @jdm

---
This change is [Reviewable](https://reviewable.io/reviews/servo/core-foundation-rs/259)

### CODE REVIEW

**Comment by jdm** [comment]
@jrmuizel Do you know things about CoreAnimation? I know nothing.

**Comment by pcwalton** [comment]
@mstange has been reviewing my Core Animation related code in Gecko, FWIW.

**Comment by mstange** [comment]
This CoreAnimation-specific parts looks straightfoward to me. What I have trouble with are the generic Objective C parts, and @jrmuizel is a better reviewer for those. For example the string conversion business and the use of `wrap_under_get_rule`, I'm not familiar with them.

File: cocoa/src/quartzcore.rs [hunk 1]
```diff
@@ -0,0 +1,1589 @@
...
+    #[inline]
+    pub fn remove_from_superlayer(&self) {
+        unsafe {
+            msg_send![self.id(), removeFromSuperlayer]
+        }
+    }
+
+    /// This is only a snapshot, unlike the corresponding Cocoa API.
+    #[inline]
+    pub fn sublayers(&self) -> Vec<CALayer> {
```
Comment (jrmuizel) [inline cocoa/src/quartzcore.rs]: Why not just return a CFArray<CALayer>?

Comment (pcwalton) [reply cocoa/src/quartzcore.rs]: Good point, made that change.

**Pull request merged**